main.go

```diff
@@ -50,21 +56,21 @@ func main() {
 	var wg sync.WaitGroup
 	for i := 1; i <= *clients; i++ {
 		wg.Add(i)
-		go transferMessages(sess, rmin, dest, &wg)
+		go transferMessages(sess, rmin, dest, &wg, *limit)
 	}
 	wg.Wait()
 
 	log.Println("all done")
 }
 
-//transferMessages loops, transferring a number of messages from the src to the dest at an interval.
-func transferMessages(theSession *session.Session, rmin *sqs.ReceiveMessageInput, dest *string, wgOuter *sync.WaitGroup) {
+// transferMessages loops, transferring a number of messages from the src to the dest at an interval.
+func transferMessages(theSession *session.Session, rmin *sqs.ReceiveMessageInput, dest *string, wgOuter *sync.WaitGroup, limit int) {
```

Comment:
Probably because of difference between pointers in strings and ints in go, but why is the dest `*string` while limit is `int`?

Reply:
the cause is that it uses goroutines, so we need to sync the count